torch/backends/_nnapi/prepare.py

```diff
@@ -0,0 +1,167 @@
+from typing import Optional, List, Tuple, Any
+
+import torch
+from torch.backends._nnapi.serializer import serialize_model
+
+class NnapiModule(torch.nn.Module):
+    """Torch Module that wraps an NNAPI Compilation.
+
+    This module handles preparing the weights, initializing the
+    NNAPI TorchBind object, and adjusting the memory formats
+    of all inputs and outputs.
+    """
+
+    comp: Optional[torch.classes._nnapi.Compilation]
+
+    def __init__(
+            self,
+            ser_model: torch.Tensor,
+            weights: List[torch.Tensor],
+            inp_mem_fmts: List[int],
+            out_mem_fmts: List[int],
+            out_templates: List[torch.Tensor]):
+        super().__init__()
+        self.ser_model = ser_model
+        self.weights = weights
+        self.inp_mem_fmts = inp_mem_fmts
+        self.out_mem_fmts = out_mem_fmts
+        self.out_templates = out_templates
+        self.comp = None
+
+    @torch.jit.export
+    def init(self):
+        assert self.comp is None
+        self.weights = [ w.contiguous() for w in self.weights ]
+        comp = torch.classes._nnapi.Compilation()
+        comp.init(self.ser_model, self.weights)
+        self.comp = comp
+        
+    def forward(self, args: List[torch.Tensor]) -> List[torch.Tensor]:
+        comp = self.comp
+        assert comp is not None
+        outs = [ torch.empty_like(out) for out in self.out_templates ]
+        
+        assert len(args) == len(self.inp_mem_fmts)
+        fixed_args = []
+        for idx in range(len(args)):
+            fmt = self.inp_mem_fmts[idx]
+            if fmt == 0:
+                fixed_args.append(args[idx].contiguous())
+            elif fmt == 1:
+                fixed_args.append(args[idx].permute(0,2,3,1).contiguous())
```

Comment:
Would it be feasible to comment here saying that `fmt == 0` is NCHW (i.e. channels first) and `fmt == 1` is NHWC or maybe used a enum for this?

Reply:
Enum might be tricky since this needs to be TorchScript-compatible, but I can certainly add a comment.